Snowflake.Data/Core/Session/ConnectionCacheManager.cs

```diff
@@ -11,9 +11,9 @@ namespace Snowflake.Data.Core.Session
     internal sealed class ConnectionCacheManager : IConnectionManager
     {
         private readonly SessionPool _sessionPool = SessionPool.CreateSessionCache();
-        public SFSession GetSession(string connectionString, SecureString password) => _sessionPool.GetSession(connectionString, password);
-        public Task<SFSession> GetSessionAsync(string connectionString, SecureString password, CancellationToken cancellationToken)
-            => _sessionPool.GetSessionAsync(connectionString, password, cancellationToken);
+        public SFSession GetSession(string connectionString, SecureString password, SecureString passcode = null) => _sessionPool.GetSession(connectionString, password, passcode);
```

Comment:
I let the cancellationToken as the last parameter of the method, so we can not have a parameter with a default value before a required parameter. Do you agree to change the order of the parameters so SecurePassword will be at the end with a default value null?

Reply:
but actually why do we need a default value of null for sercurePassword?
For instance for password we don't have a default value.
I would remove the default value.

Reply:
If we remove the default value it will be always required add a null value to call the GetSessionAsync although the user is not using a secure password. When we use the default value that makes the parameter optional

Reply:
I understand that, however what I see is that:
-  GetSession and GetSessionAsync both require password (although in many cases it is just null)
- passcode is very similar to password parameter
- for GetSession passcode is optional, for GetSessionAsync it is required

so I don't like that it is not consistent.